test/crash_course_test.exs

```diff
@@ -1,8 +1,58 @@
-defmodule CrashCourseTest do
+defmodule CrashCourseAgentTest do
   use ExUnit.Case
-  doctest CrashCourse
+  doctest CrashCourseAgent
 
-  test "greets the world" do
-    assert CrashCourse.hello() == :world
+  use ExUnit.Case
+
+  test 'launches GenServer' do
+   {status, _pid} = start_server()
+   assert status == :ok
+  end
+
+  test 'returns empty Map as a default state of GenServer' do
+    start_server()
+    assert CrashCourseAgent.get == %{}
+  end
+
+  test 'returns updated state after put action' do
+    start_server
+    CrashCourseAgent.put(:a, 1)
+    assert CrashCourseAgent.get() == %{a: 1}
+  end
+
+  test 'returns desired key value for get ' do
+    start_server()
```

Comment:
Wow. `start_server()` happened to appear for the third time and you still were not looking to DRY? :-P

[`ExUnit.Callbacks.setup/2`](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#setup/2) comes to the rescue:

```elixir
setup do
  {:ok, pid} = start_server()

  on_exit fn -> CrashCourseAgent.stop end

  [agent: pid]
end
```